extras/dht/dht.h

```diff
@@ -0,0 +1,10 @@
+#ifndef __DTH_H__
+#define __DHT_H__
+
+#include <espressif/esp_misc.h> // sdk_os_delay_us
+#include "FreeRTOS.h"
+
+bool waitForPinState(uint8_t pin, uint8_t interval, uint8_t timeout, bool expected_pin_sate, uint8_t * counter);
+bool readHumidityAndTemperature(int8_t pin, int8_t * humidity, int8_t * temperature, char ** status_message);

```

Comment:
I would prefer these function names be prefixed with something like `dht_` to keep it clear what module they're part of and reduce potential namespace collision issues..  Also, we have generally been following the lowercase-with-underscores convention for function names, rather than camelCase..